src/Microsoft.AspNetCore.HttpsPolicy/HttpsPolicyOptions.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.HttpsPolicy
+{
+    /// <summary>
+    /// Options for the HttpsPolicyMiddleware middleware
+    /// </summary>
+    public class HttpsPolicyOptions
+    {
+        /// <summary>
+        /// Initializes a new <see cref="HttpsPolicyOptions"/>
+        /// </summary>
+        public HttpsPolicyOptions()
+        {
+            HstsOptions = new HstsOptions();
+        }
+
+        /// <summary>
+        /// Whether to use HTTP Strict-Transport-Security (HSTS) on all HTTPS requests.
+        /// </summary>
+        public bool SetHsts { get; set; }
+
+        /// <summary>
+        ///  Options for using HSTS
+        /// </summary>
+        public HstsOptions HstsOptions { get; set; } // TODO should this be initialized at all? We could remove the bool if so.
```

Comment:
I'm happy with this being initialized by default (to whatever defaults we choose) and for it being null to have the meaning of SetHsts = false. That way there can't be invalid combinations (like the user initializes the setting and forgets to set the flag or viceversa). I also think we follow this pattern in other places. @Tratcher do you have anything against my proposal?